### PR TITLE
sync flink resources

### PR DESCRIPTION
**Pull request by mhoffm-aiven**
This PR aims to synchronize the flink resources ( table/job ) with the upstream API.

### CODE REVIEW

File: aiven/provider.go [hunk 1]
```diff
@@ -398,6 +398,18 @@ func validateDurationString(v interface{}, k string) (ws []string, errors []erro
 	return
 }
 
+// validateStringEnum returns a schema.ValidateFunc that ensures that a string one of a few possible values
```
Comment (ivan-savciuc) [inline aiven/provider.go:401]: Can you please check if we can reuse `validation.StringInSlice`?

Comment (mhoffm-aiven) [reply aiven/provider.go:401]: good catch we can!

File: aiven/resource_flink_table.go [hunk 2]
```diff
@@ -8,6 +8,30 @@ import (
...
+	startupModeLargestOffset  = "largest-offset"
+	startupModeGroupOffsets   = "group-offsets"
+	startupModeTimestamp      = "timestamp"
+)
+
+var (
+	aivenFlinkTableConnectorTypes    = []string{connectorTypeKafka, connectorTypeUpsertKafka}
+	aivenFlinkTableKafkaStartupModes = []string{startupModeEarliestOffset, startupModeLargestOffset, startupModeGroupOffsets, startupModeTimestamp}
+	aivenFlinkTableKafkaValueFormats = []string{formatAvro, formatAvroConfluent, formatDebeziumAvroConfluent, formatDebeziumJson, formatJson}
+	aivenFlinkTableKafkaKeyFormats   = aivenFlinkTableKafkaValueFormats
```
Comment (ivan-savciuc) [inline aiven/resource_flink_table.go:34]: Why do we need two variables with the same values?

File: aiven/resource_flink_table.go [hunk 3]
```diff
@@ -8,6 +8,30 @@ import (
...
+	formatJson                  = "json"
+
+	startupModeEarliestOffset = "earliest-offset"
+	startupModeLargestOffset  = "largest-offset"
+	startupModeGroupOffsets   = "group-offsets"
+	startupModeTimestamp      = "timestamp"
+)
+
+var (
+	aivenFlinkTableConnectorTypes    = []string{connectorTypeKafka, connectorTypeUpsertKafka}
```
Comment (ivan-savciuc) [inline aiven/resource_flink_table.go:31]: The problem with the package level variables is that someone can accidentally overwrite it's value. Can use use functions instead? 

Something like:
```golang
func getFlinkTableConnectorTypes() []string {
    return []string{connectorTypeKafka, connectorTypeUpsertKafka}
}
```

File: aiven/resource_flink_test.go [hunk 4]
```diff
@@ -13,12 +13,17 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
-func TestAccAiven_flinkBasic(t *testing.T) {
-	resourceName := "aiven_flink.bar"
-	serviceName := fmt.Sprintf("test-acc-flink-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+func TestAccAiven_flink(t *testing.T) {
+	t.Parallel()
```
Comment (ivan-savciuc) [inline aiven/resource_flink_test.go:17]: Looks like we do not need that since we use `resource.ParallelTest`

**Review by ivan-savciuc — DISMISSED** [review]

File: aiven/resource_flink_test.go (hunk 4 above)
Comment (mhoffm-aiven) [reply aiven/resource_flink_test.go:17]: I tested and it does not run in parallel without that! I think `resource.ParallelTest(tt)` only registers the test suit as parallel on `tt`.

File: aiven/resource_flink_table.go (hunk 2 above)
Comment (mhoffm-aiven) [reply aiven/resource_flink_table.go:34]: they have different semantic and only coincidentally the same values i think, i'm for separating the names

File: aiven/resource_flink_table.go (hunk 3 above)
Comment (mhoffm-aiven) [reply aiven/resource_flink_table.go:31]: :heavy_check_mark:

File: aiven/resource_flink_table.go (hunk 2 above)
Comment (mhoffm-aiven) [reply aiven/resource_flink_table.go:34]: :heavy_check_mark:

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Pull request merged**